src/plugins/namer.ts

```diff
@@ -1,37 +1,58 @@
-import { CodeModel, codeModelSchema, Property, Language} from '@azure-tools/codemodel';
+import { CodeModel, codeModelSchema, Property, Language, Metadata, Operation, OperationGroup, Parameter, ComplexSchema, ObjectSchema, ChoiceSchema, ChoiceValue, SealedChoiceSchema } from '@azure-tools/codemodel';
 import { Session, Host, startSession, Channel } from '@azure-tools/autorest-extension-base';
 import { serialize, deserialize } from '@azure-tools/codegen';
-import { values, items, length, Dictionary } from '@azure-tools/linq';
+import { values, items, length, Dictionary, keys } from '@azure-tools/linq';
+import { isNullOrUndefined } from 'util';
+import { CliCommonSchema, CliConst, LanguageType, M4Node } from '../schema';
+import { Helper } from '../helper';
 
 export class CommonNamer {
     codeModel: CodeModel
+    cliNamingSettings: CliCommonSchema.NamingConvention
+    defaultNamingSettings: CliCommonSchema.NamingConvention
+    flag: Set<Metadata>
 
     constructor(protected session: Session<CodeModel>) {
         this.codeModel = session.model;
     }
 
     async init() {
         // any configuration if necessary
+        this.cliNamingSettings = Helper.normalizeNamingSettings(await this.session.getValue("naming.cli", {}));
+        this.defaultNamingSettings = Helper.normalizeNamingSettings(await this.session.getValue("naming.default", {}));
+
```

Comment:
Will the directive settings be in readme.cli.md ?  I thought `session.getValue(something)` here the `something` should be `readme.something.md` ?

Reply:
it works as long as you put these settings under 'clicommon: ...' section. the readme.xxx.md file name doesnt matter. thx